MidasCivil_Adapter/Convert/ToBHoM/Loads/ToBarDifferentialTemperatureLoad.cs

```diff
@@ -0,0 +1,134 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System.Collections.Generic;
+using BH.oM.Adapters.MidasCivil;
+using BH.Engine.Adapter;
+using BH.oM.Structure.Loads;
+using BH.oM.Structure.Elements;
+using BH.Engine.Reflection;
+using BH.oM.Structure.SectionProperties;
+using System.Linq;
+using System.IO;
+namespace BH.Adapter.Adapters.MidasCivil
+{
+    public static partial class Convert
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static BarDifferentialTemperatureLoad ToBarDifferentialTemperatureLoad(List<string> temperatureLoad, string loadcase,
+            Dictionary<string, Loadcase> loadcaseDictionary, Dictionary<string, Bar> barDictionary, int count, string temperatureUnit, string matchingBar)
+        {
+            List<double> positions = new List<double>();
+            List<double> temperatureList = new List<double>();
+            DifferentialTemperatureLoadDirection loadDirection = new DifferentialTemperatureLoadDirection();
+            double depth = new double();
+            List<Bar> bhomAssociatedBars = new List<Bar>();
+            Bar bhomAssociatedBar = new Bar();
+            if (matchingBar.Contains(' '))
+            {
+                string[] matchingBars = matchingBar.Split(' ');
+                foreach (string matchBar in matchingBars)
+                    barDictionary.TryGetValue(matchBar, out bhomAssociatedBar);
+                bhomAssociatedBars.Add(bhomAssociatedBar);
```

Comment:
```suggestion
                foreach (string matchBar in matchingBars)
                {
                    barDictionary.TryGetValue(matchBar, out bhomAssociatedBar);
                    bhomAssociatedBars.Add(bhomAssociatedBar);
                }
```

When you have a load that is applied to mulitple objects, e.g. Bar 1 and Bar 2, you are only capturing the second id because you haven't used brackets for the `foreach` loop.

Please double check you're able to read the right elements.

![image](https://user-images.githubusercontent.com/30078587/114419144-16d98880-9bab-11eb-8ef8-115f833f778d.png)